client/allocrunner/taskrunner/plugin_supervisor_hook.go

```diff
@@ -448,18 +469,17 @@ func (h *csiPluginSupervisorHook) Stop(_ context.Context, req *interfaces.TaskSt
 	return nil
 }
 
-func (h *csiPluginSupervisorHook) kill(ctx context.Context, reason error) {
-	h.logger.Error("killing task because plugin failed", "error", reason)
+func (h *csiPluginSupervisorHook) restartTask(ctx context.Context, reason error) {
+	h.logger.Error("restarting task because plugin failed", "error", reason)
 	event := structs.NewTaskEvent(structs.TaskPluginUnhealthy)
 	event.SetMessage(fmt.Sprintf("Error: %v", reason.Error()))
 	h.eventEmitter.EmitEvent(event)
 
-	if err := h.lifecycle.Kill(ctx,
-		structs.NewTaskEvent(structs.TaskKilling).
-			SetFailsTask().
+	if err := h.lifecycle.Restart(ctx,
+		structs.NewTaskEvent(structs.TaskRestarting).
 			SetDisplayMessage(fmt.Sprintf("CSI plugin did not become healthy before configured %v health timeout", h.task.CSIPluginConfig.HealthTimeout.String())),
-	); err != nil {
-		h.logger.Error("failed to kill task", "kill_reason", reason, "error", err)
+		true); err != nil {
```

Comment:
just noting that `true` here represents `failure`, which means that each restart will count against the task's `restart.attempts` limit. seems reasonable to me - we would want it to run out of attempts and be rescheduled elsewhere, if possible (not possible with node plugins as system jobs, but hey)